jax/_src/lax/parallel.py

```diff
@@ -247,6 +247,28 @@ def _canonicalize_axis_index_groups(axis_index_groups):
     return
   return tuple(map(tuple, axis_index_groups))
 
+
+def pcopy(x, axis_name, source):
+  """Perform a collective broadcast and replicate from ``source``.
```

Comment:
Want to put this kind of thing in the docstring, as we discussed in 1:1 chat?

```python
def pcopy(x, axis_name, source):
  return psum(jnp.where(axis_index(axis_name) == source, x, jnp.zeros_like(x)), axis_name)
```

Reply:
Yes let me add that.

Reply:
done

Reply:
I modified it slightly to fit within 80 character line limit.